src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryPipeStatus.java

```diff
@@ -0,0 +1,181 @@
+package com.snowflake.kafka.connector.internal;
+
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.node.ObjectNode;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class SnowflakeTelemetryPipeStatus extends SnowflakeTelemetryBasicInfo {
+  // Offset info
+  AtomicLong processedOffset;           // processed offset + 1 (offset that is most recent in buffer)
```

Comment:
Do these variables have to be package private?  Can we change them to private?

Reply:
We are directly accessing the variables in other classes, so if changed to private, we need a function to set the value and a function to get the value, which complicate things.